distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/cometd/CometDClient.java

```diff
@@ -177,8 +179,46 @@ public void subscribe(String channel) {
     }
 
     /**
-     * Gets the list of messages received on a given channel.
+     * Returns an action url from the messages of a channel if it contains the search criteria string.
      *
+     * @param searchCriterion a string that will be searched for in the messages.
+     * @param title           the title of the action we want to find.
+     * @return The desired action url if a match for the search criterion is found in the messages.
+     * Otherwise returns an empty string.
+     * @throws Exception
+     */
+    public String findActionUrlBySearchCriterion(String searchCriterion, String title)
+            throws Exception {
+
+        List<String> messages = getAllMessages();
+
+        Optional<String> foundString = messages.stream()
+                .filter(query -> query.contains(searchCriterion))
+                .findAny();
+
+        LOGGER.debug("Data found using search criterion: " + foundString.toString());
```

Comment:
This shouldn't be calling `toString`. Rather, you should log the found/not found details:
```java
if (foundString.isPresent()) {
    LOGGER.debug("Data found using search criterion: {}", foundString.get());
} else {
    LOGGER.debug("No data found using search criterion.");
}
```